lua/lspconfig/clangd.lua

```diff
@@ -12,7 +12,27 @@ local function switch_source_header(bufnr)
   end)
 end
 
+-- https://clangd.llvm.org/extensions.html#file-status
+local function file_status_update(_, _, message, client_id)
+  local client = vim.lsp.get_client_by_id(client_id)
+  if not client then
+    vim.api.nvim_err_writeln("LSP[" .. client_id .. "] client has shut down after sending the message")
+    vim.api.nvim_command("redraw")
+  end
+  client.messages.status = { uri = message.uri, content = message.state }
+  vim.api.nvim_command('doautocmd <nomodeline> User LspProgressUpdate')
+end
+
 local root_pattern = util.root_pattern("compile_commands.json", "compile_flags.txt", ".git")
+
+local default_capabilities = vim.tbl_deep_extend('keep', require'vim.lsp.protocol'.make_client_capabilities(), {
```

Comment:
Why is this portion necessary? I don't see any modifications to the capabilities from before?

Reply:
See other comment. It was a bug with only clangd.